lib/astutils.cpp

```diff
@@ -246,7 +246,16 @@ bool isEqualKnownValue(const Token * const tok1, const Token * const tok2)
 
 bool isDifferentKnownValues(const Token * const tok1, const Token * const tok2)
 {
-    return tok1->hasKnownValue() && tok2->hasKnownValue() && tok1->values() != tok2->values();
+    if(tok1->hasKnownValue() && tok2->hasKnownValue() && tok1->values() != tok2->values()) {
+        ValueFlow::Value val1 = tok1->values().front();
+        ValueFlow::Value val2 = tok2->values().front();
+        if(val1.isIntValue() && val2.isFloatValue())
+            return val1.intvalue != val2.floatValue;
```

Comment:
When I wasn't checking for floats the test case for `x == 1 || x == 1.0` wasn't passing. I can remove this now since there isn't a test case for this anymore.